lib/auth/in_memory/indexer.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Class that provides an internal indexing over the simple data provided by
+ * the authentication configuration file for the memory backend. This allows
+ * accessing the different authentication entities through various types of
+ * keys.
+ *
+ * @class Indexer
+ */
+module.exports = class Indexer {
+    /**
+     * @constructor
+     * @param {object} authdata - the authentication config file's data
+     * @return {undefined}
+     */
+    constructor(authdata) {
+        this.accountsBy = {
+            canId: {},
+            accessKey: {},
+            email: {},
+        };
+        this.usersBy = {
+            accessKey: {},
+            email: {},
+        };
+
+        /*
+         * This may happen if the file backend is not configured for S3.
```

Comment:
You mean to delete it since we check the authdata before we create the indexer, right?